pages/api/notes/search.ts

```diff
@@ -0,0 +1,54 @@
+
+import { NextApiRequest, NextApiResponse } from "next";
+import { prisma } from "@/lib/prisma";
+import { Note } from "@prisma/client";
+
+
+export interface NotePreview {
+    id: number
+    title: string
+    content: string
+    searchTerm: string
+}
+const TITLE_LIMIT: number = 50
+const CONTENT_LIMIT: number = 100
+
+function convertToNotePreview(notes: Note[], searchTerm: string): NotePreview[] {
+    return notes.map(note => {
+        return { id: note.id, title: note.title.substring(0, TITLE_LIMIT), content: note.content.substring(0, CONTENT_LIMIT), searchTerm: searchTerm }
+    })
+}
+
+export default async function handle(req: NextApiRequest, res: NextApiResponse) {
+    var searchTerm = req.query.searchTerm as string
```

Comment:
You should replace `var` with `let`.
Also using `as string` is not the correct way for type checking, because you tell the compiler at _compile time_ that you know better and that this variable will always be of type string. But the caller could omit it or pass a `string[]`.

What you want instead is a _runtime_ check with `if (typeof searchTerm === "string")`.

Reply:
done